core/domain/lib/src/usecase/cart/add_to_cart_usecase.dart

```diff
@@ -0,0 +1,43 @@
+
+
+import '../../../domain.dart';
+import '../../repository/cartRepository.dart';
+import '../base/base_usecase.dart';
+
+class AddToCartUseCase
+    extends BaseUseCase<NetworkError,CartProductAddUseCaseParams,void> {
+  final CartRepository cartRepository;
+  AddtoCartRequest addtoCartRequest;
+
+  AddToCartUseCase(this.cartRepository,this.addtoCartRequest);
+
+  @override
+  Future<Either<NetworkError, void>> execute({required CartProductAddUseCaseParams  params}) {
+    return  cartRepository.addToCart(params.addtoCartRequest);
+    
+  }
+}
+
+class CartProductAddUseCaseParams extends Params{
+  AddtoCartRequest addtoCartRequest;
+
+  CartProductAddUseCaseParams({required this.addtoCartRequest});
+
+  @override
+  Either<AppError, bool> verify() {
+    // TODO: implement verify
+    throw UnimplementedError();
```

Comment:
please add necessary verification.
if we dont have any parameters to verify return true instead of throwing an error

Reply:
ok